tests/common.cpp

```diff
@@ -86,7 +86,7 @@ void TestMaterial::RandomTestSupportFunctionAllObjects()
 
 #ifdef DO_TEST
     for(size_t i=0; i<sObj.size(); i++)
-      supportVector[j*sObj.size() + i] = sObj[i]->support(v);
+      supportVector[j*sObj.size() + i] = sObj[static_cast<unsigned char>(i)]->support(v);
```

Comment:
I agree. operator[] should accept a size_t.
I'll correct this and revert part of this PR as a consequence.